plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundReadWrite.java

```diff
@@ -1,294 +1,331 @@
+            String encodedUri = groundServerAddress + "nodes/" + URLEncoder.encode(name, "UTF-8");
+            PostMethod post = new PostMethod(encodedUri);
+            post.setRequestHeader("Content-type", "application/json");
+            post.setRequestBody(jsonString);
+            return getNode(post);
+        } catch (IOException e) {
+            throw new GroundException(e);
+        }
 
-  private String dbName;
+    }
 
-  private String userName;
+    public Node getNode(String dbName) throws GroundException {
+        GetMethod get = new GetMethod(groundServerAddress + "nodes/" + dbName);
+        try {
+            return getNode(get);
+        } catch (IOException e) {
+            throw new GroundException(e);
+        }
+    }
 
-  private String password;
+    // method to create the NodeVersion given the nodeId and the tags
+    public NodeVersion createNodeVersion(String id, Map<String, Tag> tags, String structureVersionId, String reference,
+            Map<String, String> referenceParameters, String nodeId) throws GroundException {
+        NodeVersion nodeVersion = new NodeVersion(id, tags, structureVersionId, reference, referenceParameters, nodeId);
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonString;
+        try {
+            jsonString = mapper.writeValueAsString(nodeVersion);
+            String uri = groundServerAddress + "nodes/versions";
+            PostMethod post = new PostMethod(uri);
+            post.setRequestHeader("Content-type", "application/json");
+            post.setRequestBody(jsonString);
+            return getNodeVersion(post);
+        } catch (IOException e) {
+            throw new GroundException(e);
+        }
 
-  private String host;
+    }
 
-  public StructureVersionFactory getStructureVersionFactory() {
-    return structureVersionFactory;
-  }
+    public NodeVersion getNodeVersion(String nodeVersionId) throws GroundException {
+        GetMethod get = new GetMethod(groundServerAddress + "nodes/versions/" + nodeVersionId);
+        try {
+            return getNodeVersion(get);
+        } catch (IOException e) {
+            throw new GroundException(e);
+        }
+    }
 
-  public void setStructureVersionFactory(StructureVersionFactory svf) {
-    this.structureVersionFactory = svf;
-  }
+    public List<String> getLatestVersions(String dbName) throws GroundException {
+        // TODO Auto-generated method stub
+        GetMethod get = new GetMethod(groundServerAddress + "nodes/" + dbName + "/latest");
+        try {
+            return getList(get);
+        } catch (IOException e) {
+            throw new GroundException(e);
+        }
+    }
 
-  /**
-   * @return the sf
-   */
-  public StructureFactory getStructureFactory() {
-    return structureFactory;
-  }
+    // create edge for input Tag
+    public Edge createEdge(String name) throws GroundException {
+        Edge node = new Edge("id", name);
```

Comment:
Thanks! will fix. Most of these were inherited from old code - will clean and revise